spec.md

```diff
@@ -73,8 +73,8 @@ octal_digit = "0" … "7" .
 ### Comments
 
 There are three kinds of comments, all collectively referred to by the token `Comment`:
-  1. *Line comments* (`LineComment`s) start with the sequence `//` and end at the end of the line.
-  2. *Block comments* (`BlockComment`s) start with the sequence `/*` and end with a matching `*/`.
+  1. *Line comments* (`LineComment`s) start with the sequence `//` and end at the next `newline`.
+  2. *Block comments* (`BlockComment`s) start with the sequence `/*` and end with next occurrence of `*/`.
```

Comment:
Block comments should end on a *matching* `*/` because it's something that's very common in many "modern" languages - it's not just Rust. Scala, Swift, and Dart all do, as well. It seems to me that Go's the exception among recent languages in the way that it doesn't allow this.

Reply:
I made a comment in my commit message about this. I know it's a feature in a few languages, but I'm not sure what I think of it.
If we do specify it to be `matching`, we need to be careful about how we do it. Rusts behaviour isn't as you'd expect.
Perhaps we allow comments to be nested? So // hello */ wouldn't close it, since the */ is in another comment.
Also, out there suggestion, why don't we require the opening and closing things to have a matching number of *s. Most people wouldn't know this at first, potentially making it confusing, but I think it's a nice feature to have, since sometimes, it is convenient to not have nested comments.
I'm not sure what I think to that idea because it seems like a deviation without much benefit, but just to put it out there.